polar-core/src/vm.rs

```diff
@@ -988,12 +995,18 @@ impl PolarVirtualMachine {
         field: &Term,
         check_errors: bool,
     ) -> PolarResult<QueryEvent> {
-        let (field_name, args) = match &field.value() {
+        let (field_name, args) = match self.deref(field).value() {
             Value::Call(Predicate { name, args }) => (
                 name.clone(),
-                args.iter().map(|arg| self.deref(arg)).collect(),
+                Some(args.iter().map(|arg| self.deref(arg)).collect()),
             ),
-            _ => unreachable!("call must be a predicate"),
+            Value::String(field) => (Symbol(field.clone()), None),
+            v => {
```

Comment:
Is this path actually reachable?

Reply:
Yep, I changed the parser so a `x.y` parses as `.(x, "y")`

Reply:
To be clear, I'm asking about the catch-all path here, `v => {`

Reply:
Ohh right. I guess you could if you do `x.(1)` or something, or manually construct `.(d, 1, value)`.